src/rust/engine/src/nodes.rs

```diff
@@ -664,38 +663,40 @@ pub struct SelectTransitive {
 }
 
 impl SelectTransitive {
-  fn get_dep(&self, context: &Context, dep_subject: &Value) -> NodeFuture<Value> {
-    // TODO: This method needs to consider whether the `dep_subject` is an Address,
-    // and if so, attempt to parse Variants there. See:
-    //   https://github.com/pantsbuild/pants/issues/4020
 
-    let dep_subject_key = externs::key_for(dep_subject);
-    // After the root has been expanded, a traversal continues with dep_product == product.
-    let mut selector = self.selector.clone();
-    selector.dep_product = selector.product;
-    context.get(
-      SelectTransitive {
-        subject: dep_subject_key,
-        variants: self.variants.clone(),
-        selector: selector,
-      }
-    )
+  /**
+   * Process/hydrate single buildfile address. This also extracts its dependency addresses
```

Comment:
SelectTransitive should work for any types of keys and values, so mentioning "address"es anywhere in this change should probably be avoided.

Also, a quick comment on what the values in the tuple in the return type are would be helpful.